app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm.rb

```diff
@@ -76,4 +63,28 @@ def raw_rename
   def self.calculate_power_state(raw_power_state)
     POWER_STATES[raw_power_state] || "unknown"
   end
+
+  private
+
+  def vios?
+    description == "Virtual IO Server"
+  end
+
+  def power(params = {})
+    ext_management_system.with_provider_connection do |connection|
+      if params.key?("operation")
+        if vios?
+          connection.poweroff_vios(ems_ref, params)
+        else
+          connection.poweroff_lpar(ems_ref, params)
+        end
+      elsif vios?
+        connection.poweron_vios(ems_ref, params)
+      else
+        connection.poweron_lpar(ems_ref, params)
+      end
+    end
+  rescue IbmPowerHmc::Connection::HttpError => e
+    $ibm_power_hmc_log.error("error changing power state of #{ems_ref} with params=#{params}: #{e}")
```

Comment:
Should this re-raise the exception?  As is, this causes the method, on failure, to return `true`